tests/pmap_test.py

```diff
@@ -331,6 +331,16 @@ def testCollectivePermuteCyclicGrad(self):
     expected = onp.roll(onp.pi + onp.arange(device_count), 1)
     self.assertAllClose(ans, expected, check_dtypes=False)
 
+  @jtu.skip_on_devices("cpu", "gpu")
+  def testIssue1703(self):
```

Comment:
My favorite nit: rename to something more descriptive, like testPermuteWithGenerator

Reply:
I'll tweak the name in a follow-up.

Reply:
Done!